Content.Server/LegallyDistinctSpaceFerret/LegallyDistinctSpaceFerretSystem.cs

```diff
@@ -0,0 +1,218 @@
+
+namespace Content.Server.LegallyDistinctSpaceFerret;
+
+public sealed class LegallyDistinctSpaceFerretSystem : EntitySystem
+{
+    [Dependency] private readonly ActionsSystem _actions = default!;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly PopupSystem _popup = default!;
+    [Dependency] private readonly AudioSystem _audio = default!;
+    [Dependency] private readonly RoleSystem _role = default!;
+    [Dependency] private IChatManager _chatMan = default!;
+    [Dependency] private readonly SharedMindSystem _mind = default!;
+    [Dependency] private readonly GameTicker _ticker = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<LegallyDistinctSpaceFerretComponent, GenericAntagCreatedEvent>(OnInit);
+        SubscribeLocalEvent<LegallyDistinctSpaceFerretComponent, ComponentShutdown>(OnShutdown);
+        SubscribeLocalEvent<LegallyDistinctSpaceFerretComponent, BackflipActionEvent>(OnBackflipAction);
+        SubscribeLocalEvent<LegallyDistinctSpaceFerretComponent, EepyActionEvent>(OnEepyAction);
+        SubscribeLocalEvent<LegallyDistinctSpaceFerretComponent, InteractionAttemptFailed>(OnInteractFailed);
+        SubscribeLocalEvent<HibernateConditionComponent, ObjectiveGetProgressEvent>(OnHibernateGetProgress);
+        SubscribeLocalEvent<ConsumeNutrientsConditionComponent, ObjectiveGetProgressEvent>(OnConsumeNutrientsGetProgress);
+        SubscribeLocalEvent<LegallyDistinctSpaceFerretComponent, HungerModifiedEvent>(OnHungerModified);
+    }
+
+    private void OnInit(EntityUid uid, LegallyDistinctSpaceFerretComponent component, GenericAntagCreatedEvent args)
+    {
+        _actions.AddAction(uid, ref component.BackflipActionEntity, component.BackflipAction, uid);
+        _actions.AddAction(uid, ref component.EepyActionEntity, component.EepyAction, uid);
+
+        var mind = args.Mind;
+
+        if (mind.Session == null)
+            return;
+
+        var session = mind.Session;
+        _audio.PlayGlobal(new SoundPathSpecifier(component.RoleSfx), Filter.Empty().AddPlayer(session), false, AudioParams.Default);
+        _chatMan.DispatchServerMessage(session, Loc.GetString("legallydistinctspaceferret-role-greeting"));
+
+        _role.MindAddRole(args.MindId, new LegallyDistinctSpaceFerretRoleComponent()
+        {
+            PrototypeId = component.AntagProtoId
+        }, mind);
+
+        _role.MindAddRole(args.MindId, new RoleBriefingComponent
+        {
+            Briefing = Loc.GetString("legallydistinctspaceferret-role-briefing")
+        }, mind);
+
+        component.BrainrotEffectCanceller = new CancellationTokenSource();
+
+        Timer.SpawnRepeating(TimeSpan.FromSeconds(1), () =>
+        {
+            var mobs = new HashSet<Entity<MobStateComponent>>();
+            _lookup.GetEntitiesInRange(Transform(uid).Coordinates, component.BrainRotAuraRadius, mobs);
```

Comment:
Not that this is critical for an april fools PR, but for a future PR you can reuse the set instead of creating a new one here.

Reply:
Feel free to rip this all to shit, it is a learning wawa.